File: users-service/aws_getters.py
from botocore.exceptions import ClientError

def get_parameter(ssm_client, parameter_name):
    try:
        response = ssm_client.get_parameter(
            Name=parameter_name,
            WithDecryption=True
        )
        return response['Parameter']['Value']
    except ClientError as e:
        print(f"Error retrieving parameter {parameter_name}: {e}")
        return None

def get_secret_value(secrets_manager_client, secret_id):
    try:
        response = secrets_manager_client.get_secret_value(
            SecretId=secret_id
        )
        if 'SecretString' in response:
            return response['SecretString']
        else:
            return response['SecretBinary']
    except ClientError as e:
        print(f"Error retrieving secret {secret_id}: {e}")
        return None

File: lands/aws_getters.go
package main

import (
	"context"
	"fmt"
	"log"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
	"github.com/aws/aws-sdk-go-v2/service/ssm"
)

func getParameter(ssmClient *ssm.Client, parameterName string, ctx context.Context) string {
	withDecryption := true
	param, err := ssmClient.GetParameter(ctx, &ssm.GetParameterInput{
		Name:           aws.String(parameterName),
		WithDecryption: &withDecryption,
	})
	if err != nil {
		log.Fatalf("Failed to fetch parameter %s: %v", parameterName, err)
	}
	return aws.ToString(param.Parameter.Value)
}

func getSecretValue(secretsManagerClient *secretsmanager.Client, secretID string, ctx context.Context) ([]byte, error) {
	output, err := secretsManagerClient.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{
		SecretId: aws.String(secretID),
	})
	if err != nil {
		return nil, fmt.Errorf("unable to retrieve secret %s: %v", secretID, err)
	}

	return []byte(*output.SecretString), nil
}
